### PR TITLE
Backport BS5's color-contrast() function, use it, and deprecate color-yiq()

### LINKED ISSUE #129: Provide more utility classes for contrasting theme colors
This could be helpful in scenarios where developers provide some HTML templating, say:

```html
<footer class="bg-primary text-light mt-5">
</footer>
```

But then allow end users to customize the primary color -- developers won't necessarily know if `text-light` or `text-dark` is the right contrast. We could provide something like

```sass
.text-primary-contrast {
  color: color-yiq($primary);
}
```

Then developers could write

```html
<footer class="bg-primary text-primary-contrast mt-5">
</footer>
```

Which is similar to the automatic contrasting model for BS components like buttons

When this is done, make sure to update this section https://rstudio.github.io/bootstraplib/articles/theming.html#color-contrasting

### LINKED ISSUE #172: Consider back-porting BS5's color contrasting approach
This should make contrasting much better as well as more consistent across BS4/BS5

https://github.com/twbs/bootstrap/commit/03908ea37a55eaa44c12ce5694dddc1630c980b3

### PR DESCRIPTION
**Pull request by cpsievert**
Closes #172
Closes #129

### TODO

- [ ] Update vignettes/docs
- [x] Tests?

### CODE REVIEW

File: inst/lib/bootstrap/scss/_functions.scss [hunk 1]
```diff
@@ -71,13 +71,15 @@
 
 // Color contrast
 @function color-yiq($color, $dark: $yiq-text-dark, $light: $yiq-text-light) {
```
Comment (cpsievert) [inline inst/lib/bootstrap/scss/_functions.scss:73]: Make sure that this won't throw if yiq-text-dark/yiq-text-light are missing

Comment (cpsievert) [reply inst/lib/bootstrap/scss/_functions.scss:73]:
```suggestion
@function color-yiq($color, $dark: black, $light: white) {
```

Comment (cpsievert) [reply inst/lib/bootstrap/scss/_functions.scss:73]:
```
variable-exists("yiq-text-dark") {
@warn "color-yiq() is deprecated, use color-contrast"
}
```

File: R/bs-theme-update.R [hunk 2]
```diff
@@ -119,7 +106,9 @@ bs3_base_colors <- function(args) {
     "gray-f5" = gray(0xf5),
     "gray-f8" = gray(0xf8),
     "gray-f9" = gray(0xf9),
-    "white" = gray(0xff)
+    "white" = gray(0xff),
+    "color-contrast-dark" = gray(0x00),
+    "color-contrast-light" = gray(0xff)
```
Comment (cpsievert) [inline R/bs-theme-update.R:111]:
```suggestion
    "black" = gray(0x00)
```

**Review by schloerke — APPROVED** [review]

**Pull request merged**